src/App.js

```diff
@@ -1,19 +1,226 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import './App.css';
 
 import users from './api/users';
+import todos from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+function getTodosUserId(userId) {
+  return users.find(user => user.id === userId);
 }
 
+const todosWithUserFromFile = todos.map(todo => ({
+  ...todo,
+  user: getTodosUserId(todo.userId),
+}));
+
+export class App extends React.Component {
+  state = {
+    todosWithUser: todosWithUserFromFile,
+    todoName: '',
+    userName: '',
+    countId: todosWithUserFromFile.length,
+    noTitle: false,
+    noSelect: false,
+    maxLength: 30,
+  }
+
+  addOneId = () => {
+    this.setState(state => ({
+      countId: state.countId + 1,
+    }));
+  };
+
+  render() {
+    const {
+      todosWithUser,
+      todoName,
+      userName,
+      countId,
+      noTitle,
+      noSelect,
+      maxLength,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1>Add todo form</h1>
+
+        <form
+          action="/api/todos"
+          method="POST"
+          onSubmit={(event) => {
+            event.preventDefault();
+
+            if (!todoName && !userName) {
+              this.setState({
+                noTitle: true,
+                noSelect: true,
+              });
+
+              return;
+            }
+
+            if (!todoName) {
+              this.setState({
+                noTitle: true,
+              });
+
+              return;
+            }
+
+            if (!userName) {
+              this.setState({
+                noSelect: true,
+              });
+
+              return;
+            }
+
+            this.setState((prevState) => {
+              const newTodo = {
+                id: countId,
+                userId: prevState.userName,
```

Comment:
```suggestion
                userId: prevState.userId,
```
It would be better to rename this field in your state because it's actually user's ID, not his name